### ISSUE TITLE
flashErase() bug

### ISSUE THREAD

**Issue by DrTon** [tegesoft/flash-stm32f407#2]
Thanks for the nice library, but I have just found a bug:
- `size_t` is unsigned type, so `size` variable is always >= 0.
- check `while (size > 0)` actually means `while (size != 0)`
- if `size` given in argument != size of sector this will never give 0: `size -= flashSectorSize(sector)`

Sorry, no time to make PR, fixed code, use if you like it:

```
int flashErase(flashaddr_t address, size_t size)
{
    while (size > 0)
    {
        flashsector_t sector = flashSectorAt(address);
        int err = flashSectorErase(sector);
        if (err != FLASH_RETURN_SUCCESS)
            return err;
        address = flashSectorEnd(sector);
        size_t sector_size = flashSectorSize(sector);
        if (sector_size >= size) {
            break;
        } else {
            size -= sector_size;
        }
    }

    return FLASH_RETURN_SUCCESS;
}
```

I haven't checked other places yet, probably the same bug there.

**Comment by delaitre** [comment]
Thanks for that. This should be fixed in 00182113d5fbb07b0cc6c8170403987eace59456

**Comment by tegesoft** [comment]
Merci Jérémie !

Fred
